rust/src/worker/channel.rs

```diff
@@ -279,12 +279,8 @@ impl Channel {
                             else {
                                 match response.body().expect("failed accessing response body") {
                                     // Response has body.
-                                    Some(body_ref) => {
-                                        let _ = sender.send(Ok(Some(
-                                            body_ref
-                                                .try_into()
-                                                .expect("failed to retrieve response body"),
-                                        )));
+                                    Some(_) => {
+                                        let _ = sender.send(Ok(Some(Vec::from(&message[4..]))));
```

Comment:
I do not like that we are manually slicing `message` here. Is there a way around it? If we already have `Body`, why not sending it through? I suspect you can get `BodyRef` out of is and in happy case will not have to do extra allocation (see `DataProducerDumpRequest::Response` and some other.

Reply:
We don't have a `Body` (we don't want a `Body` as the whole thing here is accessing the data out of `Ref`s to avoid allocation) but a `BodyRef`. With the slicing I'm just removing the size (first 4 bytes) of the buffer that is no longer needed, then passing the binary to the thread waiting for the response. The receiving side will then re-interpret the buffer as a `BodyRef` and from then read everything it needs from the buffer directly.

Reply:
The thing is `4` here is a magic number. Extremely easy to change something elsewhere in the app and break it. Is there a `body_ref.as_slice()` or something? If not `deserialize_message` above or some extra function should give you the split bytes instead. It is very fragile and accidental that you remove 4 bytes at the beginning of something deep in the library.

Reply:
There is no such thing like `body_ref.as_slice()` as far as I can see. I've asked in discord tough.

We need that 32bit sized prefix for Node only. Once we don't use UnixSocket to communicate Node and Worker we can totally get rid of this.

We could maybe use a constant rather than a harcoded `4` in the meantime though.

Reply:
`deserialize_message` already removes 4 bytes. Extract removal of those bytes into a separate function and use it before calling `deserialize_message`, such that everything down from it will not have to know prefix exists.